### PR TITLE
[REF] pos*, web: use reactivity to react to changes in UI size

### PR DESCRIPTION
**Pull request by sdegueldre**
pos*: point_of_sale, pos_restaurant, pos_sale

reviously the pos would rerender the entire UI when the window was
resized. In practice, most components do not care about the size of the
UI for their rendering, and so we're doing a lot of needless work, and
also using an owl feature (deep renders) whose perennity is not certain
(we may want to remove the ability to do deep renders completely in the
future).

This commit changes the pos code to use the ui service `isSmall` field,
makes the ui service reactive so that components can subscribe to
changes inside of it with `useState`, and makes the ui service update
its internal state at most once per animation frame, instead of after
100ms without resizing, so that resizing the viewport results in a
smoother user experience.

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]

File: addons/point_of_sale/static/src/xml/Screens/PartnerListScreen/PartnerListScreen.xml [hunk 1]
```diff
@@ -6,20 +6,20 @@
...
                             <span> Save</span>
                         </t>
                     </div>
                     <div t-if="!state.detailIsShown" class="button new-customer highlight" role="img" aria-label="Add a customer"
                             t-on-click="createPartner"
                             title="Add a customer">
-                        <i t-if="env.isMobile" class="fa fa-plus"/>
-                        <t t-if="!env.isMobile"> Create</t>
+                        <i t-if="ui.isSmall" class="fa fa-plus"/>
+                        <t t-if="!ui.isSmall"> Create</t>
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/xml/Screens/PartnerListScreen/PartnerListScreen.xml:18]: Why changing the t-if above into a t-else but not this one

File: addons/point_of_sale/static/src/xml/Screens/PartnerListScreen/PartnerListScreen.xml [hunk 2]
```diff
@@ -6,20 +6,20 @@
...
                     </div>
                     <div class="button back" t-on-click="back">
-                        <i t-if="env.isMobile" class="fa fa-angle-double-left"/>
-                        <t t-if="!env.isMobile"> Discard</t>
+                        <i t-if="ui.isSmall" class="fa fa-angle-double-left"/>
+                        <t t-if="!ui.isSmall"> Discard</t>
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/xml/Screens/PartnerListScreen/PartnerListScreen.xml:22]: Same here

**Review by adgu-odoo — CHANGES_REQUESTED** [review]
Apart from the two little comments I'm ok with the PR

**Comment by adgu-odoo** [comment]
@robodoo r+

**Pull request closed without merging**